MQ2FeedMe.cpp

```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
-		if (pItem) {
-			if (!pItem->GetItemDefinition()->FoodDuration || pItem->GetItemClass() != ItemClass_Food) {
-				WriteChatf("%s:: \arThat's not food. Don't be rediculous", PLUGIN_NAME);
-				return;
-			}
-			WriteChatf("%s:: \ayFound Item: \ap%s", PLUGIN_NAME, pItem->GetName());
-			int FoodIndex = (int)Hunger.size() + 1;
-			for (const std::string& itemName : Hunger) {
-				if (ci_equals(itemName, pItem->GetName())) {
+		WriteChatf("%s:: \ayFound Item: \ap%s", PLUGIN_NAME, pItem->GetName());
+
+		auto pItemClass = pItem->GetItemClass();
+		if (pItemClass == ItemClass_Food)
+		{
+			int FoodIndex = static_cast<int>(vFoodList.size()) + 1;
+
+			for (const std::string& itemName : vFoodList)
+			{
+				if (ci_equals(itemName, pItem->GetName()))
+				{
 					WriteChatf("%s:: \ap%s \aris already on the list", PLUGIN_NAME, pItem->GetName());
 					return;
 				}
 			}
+
 			WritePrivateProfileString("Food", fmt::format("Food{}", FoodIndex), pItem->GetName(), INIFileName);
-			Hunger.push_back(pItem->GetName());
-			EzCommand("/autoinv");
-			WriteChatf("%s \agAdded\aw: \ap%s \ayto your autofeed list", PLUGIN_NAME, pItem->GetName());
-		}
-	}
-	else if (IsNumber(Arg)) {
-		FeedAt = atoi(Arg);
-		if (FeedAt < 0) FeedAt = 0;
-		else if (FeedAt > 5000) FeedAt = 5000;
-		sprintf_s(Buffer, "%d", FeedAt);
-		WritePrivateProfileString(GetCharInfo()->Name, "AutoFeed", Buffer, INIFileName);
-		WriteChatf("\ay%s\aw:: AutoFeed(\ag%s\ax).", PLUGIN_NAME, (FeedAt) ? Buffer : "\aroff");
-		if (bAnnLevels) WriteChatf("\ay%s\aw:: Current Thirst(\ag%d\ax) Hunger(\ag%d\ax)", PLUGIN_NAME, GetPcProfile()->thirstlevel, GetPcProfile()->hungerlevel);
-	}
-	else if (!_stricmp(Arg, "announceConsume")) {
-		if (bAnnConsume) {
+			vFoodList.push_back(pItem->GetName());
+		}
+		else if (pItemClass == ItemClass_Drink)
+		{
+			int DrinkIndex = static_cast<int>(vDrinkList.size()) + 1;
+
+			for (const std::string& itemName : vDrinkList)
+			{
+				if (ci_equals(itemName, pItem->GetName()))
+				{
+					WriteChatf("%s:: \ap%s \aris already on the list", PLUGIN_NAME, pItem->GetName());
+					return;
+				}
+			}
+
+			WritePrivateProfileString("Drink", fmt::format("Drink{}", DrinkIndex), pItem->GetName(), INIFileName);
+			vDrinkList.push_back(pItem->GetName());
+		}
+		else
+		{
+			WriteChatf("%s:: \arWe don't know what to do with %s.", PLUGIN_NAME, pItem->GetName());
+			return;
+		}
+
+		DoCommand("/autoinv");
+		WriteChatf("%s \agAdded\aw: \ap%s \ayto your auto%s list", PLUGIN_NAME, pItem->GetName(), (pItemClass == ItemClass_Food ? "feed" : "drink"));
+	}
+}
+
+void HandleRemoveFoodDrinkItem(const char* type, const char* item)
+{
+	if (!type)
+	{
+		return;
+	}
+
+	std::vector<std::string>* targetVector = nullptr;
+	std::string sectionName;
+
+	if (ci_equals(type, "drink"))
+	{
+		targetVector = &vDrinkList;
+		sectionName = "Drink";
+	}
+	else if (ci_equals(type, "food"))
+	{
+		targetVector = &vFoodList;
+		sectionName = "Food";
+	}
```

Comment:
this is a good approach. we could reuse it for the Add function above.

Reply:
n/m that doesn't make sense to do, add works based on item type not the param. ignore that idea!